components/server/src/oauth-server/oauth-controller.ts

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright (c) 2020 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { AuthCodeRepositoryDB } from '@gitpod/gitpod-db/lib/typeorm/auth-code-repository-db';
+import { UserDB } from '@gitpod/gitpod-db/lib/user-db';
+import { User } from "@gitpod/gitpod-protocol";
+import { log } from '@gitpod/gitpod-protocol/lib/util/logging';
+import { OAuthException, OAuthRequest, OAuthResponse } from "@jmondi/oauth2-server";
+import * as express from 'express';
+import { inject, injectable } from "inversify";
+import { Env } from "../env";
+import { clientRepository, createAuthorizationServer } from './oauth-authorization-server';
+
+@injectable()
+export class OAuthController {
+    @inject(Env) protected readonly env: Env;
+    @inject(UserDB) protected readonly userDb: UserDB;
+    @inject(AuthCodeRepositoryDB) protected readonly authCodeRepositoryDb: AuthCodeRepositoryDB;
+
+    private getValidUser = (req: express.Request, res: express.Response): User | null => {
```

Comment:
```suggestion
    private getValidUser(req: express.Request, res: express.Response): User | null {
```

Reply:
Sheesh, I blame the fog in my head earlier today. Clearly I was not as functional as I thought.

Reply:
Fixed - thanks